### PR TITLE
DAOS-2603 doc: Update internal security docs

### PR DESCRIPTION
**Pull request by kjacque**
This patch adds content to the internal design
documentation for both the control plane and
data plane security modules.

- Add control plane security design documentation on
  how certificates are used in DAOS, and on credential
  generation in the agent.
- Add data plane security design documentation on
  access control internals.

Signed-off-by: Kris Jacque <kristin.jacque@intel.com>

### CODE REVIEW

File: src/control/security/README.md [hunk 1]
```diff
@@ -1 +1,151 @@
...
+and server certificates. The admin certificate is used to protect the gRPC
+channel between the administrative node and also provide validation of the
+administrative user. The Agent certificate is used to ensure that the compute
+node belongs to the DAOS system and also provides validation of process identity
+credentials. The server certificate is used in three separate ways. First, it is
+used to protect the gRPC channel between the administrative node and the server,
+second, it protects the channel between DAOS servers, and finally, it provides
+signing functionality for data transfers between the server and compute nodes.
+
+DAOS provides a script and openssh configuration files for generating the
```
Comment (mjmac) [inline src/control/security/README.md]: Typo: openssh -> openssl, right?

File: src/control/security/README.md [hunk 2]
```diff
@@ -1 +1,151 @@
...
+the administrative interface. Likewise, we ensure that the appropriate certs are
+used by the Agent and the Server by encoding their names into the Common Name.
+
+### Protecting Administrative Channels with Certificates
+
+Administration of a DAOS cluster will be performed by an administrator using the
+dmg utility and interfacing with the control plane service for the cluster. The
+control plane service is a distributed service provided by the DAOS cluster and
+resides on each node acting as a storage node. Even though the control plane
+service is present on every node, a subset of them is relegated as the
```
Comment (mjmac) [inline src/control/security/README.md]: s/relegated/designated/, I think?

File: src/control/security/README.md (hunk 1 above)
Comment (kjacque) [reply src/control/security/README.md]: Good catch

File: src/control/security/README.md (hunk 2 above)
Comment (kjacque) [reply src/control/security/README.md]: Fixed - you're right, wrong word there. :)

File: src/security/README.md [hunk 3]
```diff
@@ -1,3 +1,82 @@
...
+
+These capabilities persist for the life of the handle, even if someone modifies
+the pool ACL or ownership. The modifications won't be used for access decisions
+until the user attempts to get a new handle for the pool. The pool handle cannot
+be revoked.
+
+In addition, the validated credential is saved in the handle data in the DAOS
+data plane during pool connect, and remains associated with the pool handle
+for its lifetime.
+This original credential is used for access decisions when the user
```
Comment (dpquigl) [inline src/security/README.md]: This either needs to be merged into the previous line or have another line added if it is supposed to be its own paragraph.

Comment (kjacque) [reply src/security/README.md]: Ah, must have missed it when I was trying to clean up the paragraph widths. It is supposed to be part of the preceding paragraph. In the final rendering it looks like it does the right thing.

**Pull request merged**